src/pdist_api.cpp

```diff
@@ -0,0 +1,135 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <miopen/common.hpp>
+#include <miopen/miopen.h>
+#include <miopen/errors.hpp>
+#include <miopen/handle.hpp>
+#include <miopen/logger.hpp>
+#include <miopen/tensor_ops.hpp>
+#include <miopen/pdist.hpp>
+
+static void LogCmdPdistBackward(const miopenTensorDescriptor_t inputDesc, const double p)
+{
+    if(miopen::IsLoggingCmd())
+    {
+        std::stringstream ss;
+        auto dtype = miopen::deref(inputDesc).GetType();
+
+        if(dtype == miopenDouble)
+        {
+            ss << "pdistbwdfp64";
+        }
+        else if(dtype == miopenFloat)
+        {
+            ss << "pdistbwdfp32";
+        }
+        else if(dtype == miopenHalf)
+        {
+            ss << "pdistbwdfp16";
+        }
+        else if(dtype == miopenBFloat16)
+        {
+            ss << "pdistbwdbfp16";
+        }
+
+        ss << "-dims ( ";
+        for(unsigned long i : miopen::deref(inputDesc).GetLengths())
```

Comment:
uint64_t

Reply:
both uint64_t and size_t are fine but dont use unsigned long (which does not guarantee 64 bit or big enough to contain the size of the biggest object your system can handle)